src/Security/Security.php

```diff
@@ -1273,8 +1226,14 @@ public static function database_is_ready()
      */
     public static function clear_database_is_ready()
     {
-        self::$database_is_ready = null;
-        self::$force_database_is_ready = null;
+        $toClear = [
+            Member::class,
+            Group::class,
+            Permission::class,
+        ];
+        foreach ($toClear as $class) {
+            DB::clear_database_is_ready($class);
```

Comment:
this feels like a subtle, but still significant change in behaviour. When before this would completely reset the `ready` state of the database, we are now only resetting the ready state of some particular classes. As such, I think this should call `DB::clear_database_is_ready()` (no argument) to completely reset the state and so that it has the equivalent behaviour to before.

Reply:
The way I think of it, the only reason it completely set the "ready" state of the database is because the "ready" state only recorded classes relevant to Security anyway - there was no way to record a "ready" state for any other classes/tables.
I'll make this change if you want me to, but I think it is more logically consistent for `Security::clear_database_is_ready` to only clear the ready state of classes relevant to `Security` - i.e. the classes that have their ready state set by `Security::database_is_ready`

Reply:
I think have the list of 3 classes is correct, because that does mimick the previous behaviour because there was no way that a class other than Member, Group or Permission would be added to the $database_is_ready array

However it also previously had `ClassInfo::dataClassesFor(Member::class)` meaning subclasses of Member (though looks like it was assumed Group and Permission would never be subclassed, which seems fair enough, we've make a similar assumption about Folder in the assets module)

So you'll need to update this array to include all Member subclasses

Reply:
> So you'll need to update this array to include all Member subclasses

`DB::clearDatabaseIsReady()` has an optional `$clearFullHierarchy` parameter which defaults to true, which the adds `ClassInfo::dataClassesFor($class)` to the list of classes to be cleared - so subclasses of `Member` are already cleared with the current implementation.